meson.build

```diff
@@ -0,0 +1,216 @@
+  )
+  systemd_generator_dir =  systemd.get_variable(
+    pkgconfig: 'systemdsystemgeneratordir'
+  )
+  subdir('systemd')
+  udev_dir = udev.get_variable(
+    pkgconfig: 'udevdir', default_value: '/usr/lib/udev'
+  )
+  install_data(
+    'udev/66-azure-ephemeral.rules',
+    install_dir: join_paths(udev_dir, 'rules.d'),
+    install_tag: 'systemd'
+  )
+
+  # Must generate systemd templates in root mesonbuild, because nested
+  # systemd/meson.build results in builddir @OUTPUT@ macro being doubly-nested
+  # paths
+  systemd_generators = run_command(find, 'systemd', '-name', '*generator*.tmpl', check: true).stdout().strip().split('\n')
+  foreach template: systemd_generators
+    custom_target(
+      input: template,
+      output: '@BASENAME@',
+      command: [render_tmpl, '@INPUT@', join_paths(meson.current_build_dir(), '@OUTPUT@')],
+      install: true,
+      install_dir: systemd_generator_dir,
+      install_tag: 'systemd'
+    )
+  endforeach
+
+  systemd_templates = run_command(find, 'systemd', '-not', '-name', '*generator*', '-name', '*.tmpl', check: true).stdout().strip().split('\n')
+  foreach template: systemd_templates
+    custom_target(
+      input: template,
+      output: '@BASENAME@',
+      command: [render_tmpl, '@INPUT@', join_paths(meson.current_build_dir(), '@OUTPUT@')],
+      install: true,
+      install_dir: systemd_unit_dir,
+      install_tag: 'systemd'
+    )
+  endforeach
+endif
+
+# Schema files
+json_schemas = run_command(find, 'cloudinit', '-name', '*.json', check: true).stdout().strip().split('\n')
+python.install_sources(json_schemas, subdir: join_paths('cloudinit', 'config', 'schemas'))
+
+custom_target(
+  input: 'config/cloud.cfg.tmpl',
+  output: 'cloud.cfg',
+  command: [render_tmpl, '--is-yaml', '@INPUT@', join_paths(meson.current_build_dir(), '@OUTPUT@')],
+  install: true,
+  install_dir: join_paths(sysconfdir, 'cloud'),
+  install_tag: 'config'
+)
+
+
+# Docs and Examples:
+subdir('doc/examples')
+subdir('doc/examples/seed')
+mandir = get_option('mandir')
+foreach man_page : ['cloud-init.1', 'cloud-id.1', 'cloud-init-per.1']
+  install_data(
+    'doc/man/' + man_page,
+    install_dir: join_paths(mandir,  'man1'),
+    install_tag: 'docs'
+  )
+endforeach
+
+# Prerequisite for using install_deps or install_test_deps:
+# python3 -m venv ../.venv
+# source ../.venv/bin/activate
+# meson setup ../builddir -Dinit_system=systemd -Ddistro=ubuntu
+# meson compile -C ../builddir install_deps
+# meson compile -C ../builddir install_test_deps
+run_target(
+  'install_deps',
+  command: [python, '-m', 'pip', 'install', '-r', meson.project_source_root() / 'requirements.txt'],
+)
+run_target(
+  'install_test_deps',
+  command: [python, '-m', 'pip', 'install', '-r', meson.project_source_root() / 'test-requirements.txt'],
+)
+
+# CLI: meson test -C ../builddir py3 -v
+test(
+  'py3',
+  python,
+  args: ['-m', 'pytest', meson.project_source_root() / 'tests' / 'unittests'],
+  timeout: 180,
```

Comment:
Timeouts are arbitrary - a meson test wrapper is not the right place to set one. Lets set this to 0 or -1 to disable timeouts.